### PR TITLE
Fix Filter / where clause without column names is removed in optimization pass

### LINKED ISSUE #220: Filter / where clause without column names is removed in optimization pass
**Describe the bug**

**To Reproduce**
CREATE EXTERNAL TABLE csv (price INT) 
STORED AS CSV
LOCATION './tests/example.csv';

`./tests/example.csv`:
```
1
2
3
```
```sql
CREATE EXTERNAL TABLE example (price INT) 
STORED AS CSV
LOCATION './tests/example.csv';
> select * from example where 1=2 ;
+-------+
| price |
+-------+
| 1     |
| 2     |
| 3     |
+-------+
```

**Expected behavior**
Should return an empty result set. When inspecting the optimized plan, the filter clause is removed (without changing the rest of the plan).

**Additional context**
The problem goes away when removing `FilterPushDown` so this is likely the offender

### PR DESCRIPTION
**Pull request by Dandandan**
# Which issue does this PR close?

Closes #220

 # Rationale for this change
DataFusion gives wrong results for filters that don't reference any column 

# What changes are included in this PR?

Workarounds the issue by not pushing down predicates that do not reference any column (1=2, FALSE, TRUE). 
As we should be able to remove these anyway with constant folding and removing/replacing `WHERE FALSE`/ `WHERE TRUE`, I think this is a good option.

# Are there any user-facing changes?

No, results should be correct now.

### CODE REVIEW

File: datafusion/src/optimizer/filter_push_down.rs [hunk 1]
```diff
@@ -368,6 +381,7 @@ fn optimize(plan: &LogicalPlan, mut state: State) -> Result<LogicalPlan> {
                     new_filters.push(filter_expr.clone());
                 }
             }
+            println!("{:?}", &state);
```
Comment (houqp) [inline datafusion/src/optimizer/filter_push_down.rs:384]: forgot to delete this? :P

Comment (Dandandan) [reply datafusion/src/optimizer/filter_push_down.rs]: Thanks, yes, old fashioned debugging 😆

Comment (houqp) [reply datafusion/src/optimizer/filter_push_down.rs]: lol, we are in the same camp, I basically only use print for debugging except for rare kernel bugs :D

**Review by houqp — APPROVED** [review]

**Review by jorgecarleitao — COMMENTED** [review]
Looks great, ready to sail! :)

**Comment by alamb** [comment]
Thanks @Dandandan

**Pull request merged**